### PR TITLE
Reimplement unamed-task rule as name[missing]

### PR DESCRIPTION
**Pull request by ssbarnea**
This reimplementation will allow us to implement closely related extra checks such [`name[casing]`](https://english.stackexchange.com/questions/413690/is-casing-a-valid-alternative-to-capitalization#:~:text=Standard%20terminology%20in%20the%20world,it%2C%20so%20that%20the%20typesetting), `name[imperative]`, `name[unique]`, `name[suffix]`.

Related: #2171 #2170 #2169

### CODE REVIEW

File: src/ansiblelint/rules/unnamed_task.py [hunk 1]
```diff
@@ -1,48 +0,0 @@
-"""Implementation of unnamed-task rule."""
-
-# Copyright (c) 2016 Will Thames <will@thames.id.au>
```
Comment (cognifloyd) [inline src/ansiblelint/rules/unnamed_task.py:3]: Do we need to include the copyright notice in the new file?

File: src/ansiblelint/rules/name.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    severity = "MEDIUM"
+    tags = ["idiom"]
+    version_added = "historic"
+
+    def matchtask(
+        self, task: Dict[str, Any], file: "Optional[Lintable]" = None
+    ) -> Union[bool, str, MatchError]:
+        if not task.get("name"):
+            return self.create_matcherror(
+                linenumber=task["__line__"], tag="name[missing]", filename=file
```
Comment (cognifloyd) [inline src/ansiblelint/rules/name.py:31]: Let's use the LINE_NUMBER_KEY constant in ansiblelint.utils.LINE_NUMBER_KEY

Comment (cognifloyd) [reply src/ansiblelint/rules/name.py:31]: Also, this bypasses some of the enrichment that happens in `matchtasks()` like `details` and `match.task`.
https://github.com/ansible/ansible-lint/blob/a5f0d521e404ae8839d5879b20fb44200aeca1f9/src/ansiblelint/rules/__init__.py#L162-L177

I'm not sure where we should add that. Maybe `matchtasks()` should be refactored to update `details` if it is not set, and add `match.task`.

**Review by cognifloyd — COMMENTED** [review]
Dang. Sorry I didn't review before this was merged. I have a few questions to address in a follow-up.

File: src/ansiblelint/rules/unnamed_task.py (hunk 1 above)
Comment (ssbarnea) [reply src/ansiblelint/rules/unnamed_task.py:3]: Not really needed as it does not have a legal value, also the new one is practically a rewrite, that is why I did not include it.

File: src/ansiblelint/rules/name.py (hunk 2 above)
Comment (ssbarnea) [reply src/ansiblelint/rules/name.py:31]: I was aware about missing details but I was not sure what to include in them. The reality is that the generic "name" description is no longer relevant for various violations.

Probably we need to add some longer descriptions but we need to write the text for them.

**Pull request merged**